### PR TITLE
mvenherak: parking lot, exceptions

### PR DESCRIPTION
**Pull request by venherak**

### CODE REVIEW

File: src/main/java/parkinglot/ParkingLot.java [hunk 1]
```diff
@@ -61,74 +70,107 @@ public String parkVehicleOnParkingLot(AbstractVehicle vehicle) {
...
-            spotToClean = i;
-            if (parkingSpotsTicketsMap.get(Integer.toString(spotToClean)).getParkingTicketNumber()
-                    .equalsIgnoreCase(parkingTicketNumber)) {
+        if (!vehicles.contains(vehicle)) {
+            throw new VehicleNotFoundOnParkingLotException("Vehicle has not been found on parking lot.\n");
+        }
 
-                parkingSpots.get(spotToClean).unparkVehicleFromParkingSpot(vehicle);
-                vehicles.set(spotToClean, null);
+        boolean ticketFound = false;
```
Comment (nmelnyk-corevalue) [inline src/main/java/parkinglot/ParkingLot.java]: isTicketFound (convention)

File: src/main/java/parkinglot/ParkingLot.java [hunk 2]
```diff
@@ -61,74 +70,107 @@ public String parkVehicleOnParkingLot(AbstractVehicle vehicle) {
...
+                break;
             }
         }
-        return -1;
+
+        return freeSpot;
     }
 
-    public String getParkingTicketNumber(AbstractVehicle vehicle) {
+    public String getParkingTicketNumber(AbstractVehicle vehicle) throws VehicleNotFoundOnParkingLotException {
```
Comment (nmelnyk-corevalue) [inline src/main/java/parkinglot/ParkingLot.java]: throws VehicleNotFoundOnParkingLotException -> redundant

File: src/main/java/parkinglot/ParkingLotTest.java [hunk 3]
```diff
@@ -22,45 +30,118 @@ public static void main(String[] args) throws InterruptedException {
...
+        } catch (ParkingLotIsFullException e) {
+            System.out.println(e.getMessage());
+        }
         TimeUnit.SECONDS.sleep(timeout);
+
+
         // unpark vehicle
-        parkingTicketNumber = parkingLot.getParkingTicketNumber(motorcycle1);
-        System.out.println(parkingLot.unparkVehicleFromParkingLot(motorcycle1, parkingTicketNumber));
+        try {
```
Comment (nmelnyk-corevalue) [inline src/main/java/parkinglot/ParkingLotTest.java]: maybe it is better to encapsulate those try-catch blocks within the methods, not to show it in ParkingLotTest

**Review by nmelnyk-corevalue — APPROVED** [review]
fix the comments pliz and merge the changes

File: src/main/java/parkinglot/ParkingLotTest.java (hunk 3 above)
Comment (venherak) [reply src/main/java/parkinglot/ParkingLotTest.java]: fixed

File: src/main/java/parkinglot/ParkingLot.java (hunk 2 above)
Comment (venherak) [reply src/main/java/parkinglot/ParkingLot.java]: fixed

File: src/main/java/parkinglot/ParkingLot.java (hunk 1 above)
Comment (venherak) [reply src/main/java/parkinglot/ParkingLot.java]: fixed

**Pull request merged**